docs: explain using refs in a config file (#1364)
Redocly/redocly-cli: docs/configuration/index.md

The first match takes priority when a URL matches multiple patterns.
Therefore, only the header from the first match is used in the request.

### Split up the configuration file

As your config file grows, you may want to split it into multiple parts.

> That is possible by using references in a config similar to how they are used in OpenAPI descriptions:

Reviewer: ```suggestion
Splitting a config file is possible by using references in a config similar to how they are used in OpenAPI descriptions:
```

```yaml
extends:
  - recommended

theme:
  openapi:
    $ref: ./openapi-theme.yaml
  mockServer:
    $ref: ./mockserver.yaml
```

{% admonition type="attention" %}

> When using the `push` command with `$ref`s in a config file, all referenced files should explicitly upload using the `--files` option

Reviewer: ```suggestion
When using the `push` command with a config file that includes `$ref`s , all referenced files are explicitly uploaded using the `--files` option.
```

Author: @HCloward would the following work instead?

```suggestion
When using the `push` command with a config file that includes `$ref`s, all referenced files should be explicitly uploaded using the `--files` option.
```

Reviewer: yep, that works - why "should" though -  is there a possibility it won't work for some reason?

Author: Not really. It means that users should manually specify all the parts that need to be uploaded in the `--files` option. Do you think "are" fits here better?

Author: Ok, thanks for the clarification!
